### PR TITLE
feature/#3144 3145 add, edit and delete services on pricing page

### PR DESCRIPTION
**Pull request by nazarf1**

### CODE REVIEW

**Review by NewArtMask — APPROVED** [review]

File: src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-add-service-popup/ubs-admin-tariffs-add-service-popup.component.ts [hunk 1]
```diff
@@ -1,32 +1,69 @@
...
+      .pipe(takeUntil(this.destroy))
+      .subscribe(() => {
+        this.loadingAnim = false;
+        this.dialog.closeAll();
+      });
+  }
+
+  ngOnDestroy() {
+    this.destroy.unsubscribe();
+  }
```
Comment (nazarhlyvka) [inline src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-add-service-popup/ubs-admin-tariffs-add-service-popup.component.ts:68]: add this.destroy.next() before

File: src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-delete-popup/ubs-admin-tariffs-delete-popup.component.ts [hunk 2]
```diff
@@ -1,8 +1,33 @@
...
+      .deleteService(receivedData.bagData.id)
+      .pipe(takeUntil(this.destroy))
+      .subscribe(() => {
+        this.dialog.closeAll();
+      });
+  }
+
+  ngOnDestroy() {
+    this.destroy.unsubscribe();
+  }
```
Comment (nazarhlyvka) [inline src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-delete-popup/ubs-admin-tariffs-delete-popup.component.ts:32]: add this.destroy.next() before

File: src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs.component.ts [hunk 3]
```diff
@@ -1,19 +1,48 @@
...
+export class UbsAdminTariffsComponent implements OnInit, OnDestroy {
+  minAmountOfBigBags: number;
+  locations: Locations;
+  isLoadBar: boolean;
+  isFetching = false;
+  selectedLocationId;
+  bags: Bag[];
+  private destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
+  private destroy: Subject<boolean> = new Subject<boolean>();
+  public currentLanguage: string;
```
Comment (nazarhlyvka) [inline src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs.component.ts:27]: two types of destroy? i guess one is enough.

Comment (nazarf1) [reply src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs.component.ts:26]: done

File: src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-delete-popup/ubs-admin-tariffs-delete-popup.component.ts (hunk 2 above)
Comment (nazarf1) [reply src/app/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-delete-popup/ubs-admin-tariffs-delete-popup.component.ts:33]: done

**Review by nazarhlyvka — APPROVED** [review]

**Review by TarasKuzma1999 — APPROVED** [review]

**Review by Bishko86 — APPROVED** [review]

**Review by Flash5482 — APPROVED** [review]

**Review by KhrystynaPr — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Pull request merged**